test/functional/shada/marks_spec.lua

```diff
@@ -269,4 +269,19 @@ describe('ShaDa support code', function()
     eq('Vim(normal):E20: Mark not set', exc_exec('normal! `a'))
     eq('Vim(normal):E20: Mark not set', exc_exec('normal! `.'))
   end)
+
+  it('updates deleted marks for entire buffer', function ()
+    nvim_command('edit ' .. testfilename)
+
+    nvim_command('mark a')
+    feed('ggifoobar<esc>')
+    nvim_command('wshada')
+    nvim_command('normal! `a`.')
+    nvim_command('delmarks!')
+    nvim_command('wshada')
+
+    reset()
+    eq('Vim(normal):E20: Mark not set', exc_exec('normal! `a'))
+    eq('Vim(normal):E20: Mark not set', exc_exec('normal! `.'))
+  end)
```

Comment:
This test always succeeds because it doesn't edit the file after `reset()`.

If it edits the file, however, then it fails, so somehow it still isn't working.
```diff
diff --git a/test/functional/shada/marks_spec.lua b/test/functional/shada/marks_spec.lua
index 47750e52c..fa6133b98 100644
--- a/test/functional/shada/marks_spec.lua
+++ b/test/functional/shada/marks_spec.lua
@@ -281,6 +281,7 @@ describe('ShaDa support code', function()
     nvim_command('wshada')
 
     reset()
+    nvim_command('edit ' .. testfilename)
     eq('Vim(normal):E20: Mark not set', exc_exec('normal! `a'))
     eq('Vim(normal):E20: Mark not set', exc_exec('normal! `.'))
   end)
```

Reply:
I wrote this test by following the one [for `delmarks`](https://github.com/MariaSolOs/neovim/blob/3c78d17f7636dd57f5702f05d0492535c1a459b5/test/functional/shada/marks_spec.lua#L273). Does this mean that the other test has the same issue?

Reply:
Oh wait you just actually modified the test in `master`.